### PR TITLE
MAINTAINERS: Add Laird Connectivity platform

### PR DESCRIPTION
**Pull request by nordicjm**
Adds maintainers for Laird Connectivity boards

### CODE REVIEW

**Comment by nashif** [comment]
Please consider adding an area for you boards in the MAINTAINERS.yml file and add responsible people there, CODEOWNERS file is on the path of deprecation.

**Comment by lairdjm** [comment]
> Please consider adding an area for you boards in the MAINTAINERS.yml file and add responsible people there, CODEOWNERS file is on the path of deprecation.

Added to maintainers file instead. Should I also remove entries from CODEOWNERS or leave that as-is?

**Review by nashif — APPROVED** [review]

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -1157,6 +1157,20 @@ Base OS:
     labels:
         - "area: Base OS"
 
+Laird Connectivity platforms:
+    status: maintained
+    maintainers:
+        - greg-leach
```
Comment (nashif) [inline MAINTAINERS.yml:1163]: please add just one maintainer and everyone else as collaborators. We are in the process of cleaning up the file to follow this rule.

**Review by nashif — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (lairdjm) [reply MAINTAINERS.yml]: Done

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]

**Pull request merged**